src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -5,32 +5,69 @@
  */
 
 #include "z_en_honotrap.h"
+#include "assets/objects/gameplay_keep/gameplay_keep.h"
+#include "assets/objects/gameplay_dangeon_keep/gameplay_dangeon_keep.h"
 
 #define FLAGS (ACTOR_FLAG_10)
 
-#define THIS ((EnHonotrap*)thisx)
+#define HONOTRAP_AT_ACTIVE (1 << 0)
+#define HONOTRAP_AC_ACTIVE (1 << 1)
+#define HONOTRAP_OC_ACTIVE (1 << 2)
+
+typedef enum {
+    /* 0 */ HONOTRAP_EYE_OPEN,
+    /* 1 */ HONOTRAP_EYE_HALF,
+    /* 2 */ HONOTRAP_EYE_CLOSE,
+    /* 3 */ HONOTRAP_EYE_SHUT
+} EnHonotrapEyeState;
 
 void EnHonotrap_Init(Actor* thisx, PlayState* play);
 void EnHonotrap_Destroy(Actor* thisx, PlayState* play);
 void EnHonotrap_Update(Actor* thisx, PlayState* play);
 void EnHonotrap_Draw(Actor* thisx, PlayState* play);
 
-void func_8092EBDC(EnHonotrap* this, PlayState* play);
-void func_8092ECF0(EnHonotrap* this, PlayState* play);
-void func_8092EDD8(EnHonotrap* this, PlayState* play);
-void func_8092EE1C(EnHonotrap* this, PlayState* play);
-void func_8092EE90(EnHonotrap* this, PlayState* play);
-void func_8092EF98(EnHonotrap* this, PlayState* play);
-void func_8092F074(EnHonotrap* this, PlayState* play);
-void func_8092F0B8(EnHonotrap* this, PlayState* play);
-void func_8092F10C(EnHonotrap* this, PlayState* play);
-void func_8092F208(EnHonotrap* this, PlayState* play);
-void func_8092F3D8(EnHonotrap* this, PlayState* play);
-void func_8092F5EC(EnHonotrap* this, PlayState* play);
-void func_8092F7BC(EnHonotrap* this, PlayState* play);
-void func_8092F878(EnHonotrap* this, PlayState* play);
-
-#if 0
+void EnHonotrap_GetNormal(Vec3f* normal, Vec3f* vec);
+void EnHonotrap_InitEye(EnHonotrap* this, PlayState* play);
+void EnHonotrap_InitFlame(EnHonotrap* this, PlayState* play);
+void EnHonotrap_InitFlameGroup(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeIdle(EnHonotrap* this);
+void EnHonotrap_EyeIdle(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeOpen(EnHonotrap* this);
+void EnHonotrap_EyeOpen(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeAttack(EnHonotrap* this);
+void EnHonotrap_EyeAttack(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeClose(EnHonotrap* this);
+void EnHonotrap_EyeClose(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeIdle2(EnHonotrap* this);
+void EnHonotrap_EyeIdle2(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeOpen2(EnHonotrap* this);
+void EnHonotrap_EyeOpen2(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeAttack2(EnHonotrap* this);
+void EnHonotrap_EyeAttack2(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupEyeClose2(EnHonotrap* this);
+void EnHonotrap_EyeClose2(EnHonotrap* this, PlayState* play);
+void EnHonotrap_SetupFlame(EnHonotrap* this);
+void EnHonotrap_Flame(EnHonotrap* this, PlayState* play);
```

Comment:
I think I'd like to use this pattern, but if I do so it would make more sense to change `EnHonotrap_Draw` -> `EnHonotrap_Eye_Draw` to keep it in sync with oot naming, which is contrary to what was discussed in discord. MM simply does not have the equivalent function as it was refactored out.

Reply:
Rest of the PR looks good, just this single action func. I think `EnHonotrap_Flame` isn't really a descriptive enough action, what do you think of the suggestions `EnHonotrap_FlameGrow` or `EnHonotrap_FlameScaleUp`?

Reply:
I'll go with `EnHonotrap_FlameGrow`